### ISSUE TITLE
Old event-source-mappings with disabled dynamodb streams are also invoking the lambda function

### ISSUE THREAD

**Issue by TanviAgg** [localstack/localstack#2561]
# Type of request
[x] bug report
[ ] feature request

# Detailed description
I am using a lambda function with a dynamodb stream (async). After creating a dynamodb table (with stream enabled), I create an event-source-mapping between the stream and lambda function. My use-case requires me to delete and re-create the same dynamodb table and create a new event source mapping (a few times).

## Expected behavior
When a table is deleted, the stream associated with that table gets disabled. When I recreate a table with stream, the arn would be different, and hence, I would expect the old event-source-mappings to not trigger the lambda function.

## Actual behavior
For any entry in the dynamodb table, the lambda function was getting invoked n times, where n is the number of event-source-mappings created till date, including those with disabled streams.

# Steps to reproduce
Step 1: Create a lambda function.
Step 2: Create a dynamodb table with stream.
Step 3: Add an event source mapping for the stream and lambda function created above.
  // Add some items to dynamo table, this triggers lambda
Step 4: Delete the table and recreate it (with the same name) and stream enabled (this time it has a different arn)
Step 5: Create a new event source mapping for the table and lambda function.
  // Add some items to dynamodb table
Step 6: Repeat steps 4 and 5 n times.

Now, whenever we add items to the dynamodb table, it triggers lambda n number of times, depending on the number of event source mappings that exist.

## Command used to start LocalStack
```
docker run -it -e SYNCHRONOUS_DYNAMODB_EVENTS=False -e LAMBDA_EXECUTOR=docker -e -p 4566:4566 --rm --privileged --name $(LOCAL_DDB_NAME) -p 8080-8081:8080-8081 -p 4567-4620:4567-4620 -p 12121:12121 -v "$(DIR)/release/tmp/localstack:/tmp/localstack" -v "/var/run/docker.sock:/var/run/docker.sock" -e DOCKER_HOST="unix:///var/run/docker.sock" -e HOST_TMP_FOLDER="$(DIR)/release/tmp/localstack" -d "localstack/localstack"
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
To create the table with stream enabled
```
awslocal dynamodb create-table --table-name local-users \
--attribute-definitions 'AttributeName=a1,AttributeType=S' \
--key-schema 'AttributeName=a1,KeyType=HASH' \
--billing-mode PAY_PER_REQUEST \
--stream-specification  'StreamEnabled=true,StreamViewType=NEW_AND_OLD_IMAGES'
```

To create the lambda function
```
awslocal lambda create-function --function-name f1 \
--zip-file fileb://main.zip \
--handler main \
--environment "Variables={API_HOST=http://host.docker.internal:8085}" \
--runtime 'go1.x' \
--role r
```

To get the stream arn
```
STREAM_ARN=$(awslocal dynamodb describe-table --table-name local-users \
--query 'Table.LatestStreamArn' 
--output text)
```

To create the event-source-mapping
```
awslocal lambda create-event-source-mapping --function-name f1 \
--event-source-arn $STREAM_ARN \
--batch-size 1 \
--maximum-retry-attempts 1 \
--starting-position TRIM_HORIZON
```

To ascertain whether this expectation is correct, I tried to find more information about AWS streams, and it says [here](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_DeleteTable.html) that when a table is deleted the associated stream automatically gets disabled. Please let me know if more details are required. 
This is a really helpful library by the way, thank you!

**Comment by whummer** [comment]
Thanks for reporting this issue @TanviAgg . This should be fixed in #2580 - can you please give it a try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks!

**Comment by TanviAgg** [comment]
Hi, I tried this again, but the stream and event-source-mappings are still not getting removed.

### Steps to check:
Step 1: Create the table `local_users` with stream enabled.
Step 2: Create the lambda function `f1`.
Step 3: Create an event-source-mapping between the stream for this table and lambda function.
Step 4: Delete the table.
Step 5: Check if stream and event-source-mappings exist. (They do)

### Console output:
1. Describe table after calling delete says that the table does not exist.
```
$ awslocal dynamodb describe-table --table-name local_users
An error occurred (ResourceNotFoundException) when calling the DescribeTable operation: Cannot do operations on a non-existent table
```
2. But listing stream and event-source-mapping shows the corresponding entities as active.
```
$ awslocal dynamodbstreams list-streams
{
    "Streams": [
        {
            "StreamArn": "arn:aws:dynamodb:ap-southeast-1:000000000000:table/local_users/stream/2020-06-23T07:26:50.867",
            "TableName": "local_users",
            "StreamLabel": "2020-06-23T07:26:50.867"
        }
    ]
}

$ awslocal lambda list-event-source-mappings
{
    "EventSourceMappings": [
        {
            "UUID": "64dc1464-1cba-4b5c-a077-b72792e40b97",
            "BatchSize": 1,
            "EventSourceArn": "arn:aws:dynamodb:ap-southeast-1:000000000000:table/local_users/stream/2020-06-23T07:26:50.867",
            "FunctionArn": "arn:aws:lambda:ap-southeast-1:000000000000:function:f1",
            "LastModified": 1592897219.0,
            "LastProcessingResult": "OK",
            "State": "Enabled",
            "StateTransitionReason": "User action"
        }
    ]
}
```
So, as I understand, this issue is not fixed yet.

**Comment by whummer** [comment]
@TanviAgg apologies for the delay - this is now fixed as part of #3395 . The change will get merged and released shortly. Thanks
